src/tasks/sentry/tasksentry.cpp

```diff
@@ -21,6 +28,42 @@ TaskSentry::TaskSentry(const QByteArray& envelope) : Task("TaskSentry") {
 TaskSentry::~TaskSentry() { MVPN_COUNT_DTOR(TaskSentry); }
 
 void TaskSentry::run() {
+  if (!isCrashReport()) {
+    // This is not a crash report, but a ping
+    if (SettingsHolder::instance()->gleanEnabled()) {
+      // In case telemetry is enabled, send the request
+      logger.info() << "Sending Sentry-Ping";
+      sendRequest();
+    } else {
+      logger.info() << "Dropping Sentry-Ping: No Consent";
+      // Throw it away.
+      emit completed();
+    }
+    return;
+  }
+
+  // This is a crash report - Make sure we have consent, or ask the user for it
+  auto consent = SentryAdapter::instance()->hasCrashUploadConsent();
+
+  if (consent == SentryAdapter::UserConsentResult::Pending) {
+    // The consent screen was now triggered;
+    // exit here and connect to the change event to re-run.
+    connect(SentryAdapter::instance(), &SentryAdapter::userConsentChanged, this,
```

Comment:
can be the dialog close? Can you get blocked here waiting for the user consent?

Reply:
Can only be closed by choosing an option.